pennylane/qchem/tapering.py

```diff
@@ -538,3 +538,77 @@ def taper_hf(generators, paulixops, paulix_sector, num_electrons, num_wires):
         tapered_hartree_fock.append(0)
 
     return np.array(tapered_hartree_fock).astype(int)
+
+
+def _is_commuting_obs(ham_a, ham_b, wire_map=None):
+    r"""Check for commutivity between two Pauli observables.
+
+    Args:
+        ham_a (Hamiltonian): first observable
+        ham_b (Hamiltonian): second observable
+
+    Returns:
+        Bool: representing whether ham_a and ham_b commutes or not.
+    """
+    for op1 in ham_a.ops:
+        for op2 in ham_b.ops:
+            if not qml.grouping.is_commuting(op1, op2, wire_map):
+                return False
+    return True
+
+
+def taper_excitations(generators, paulixops, paulix_sector, singles, doubles):
+    r"""Transform excitations with a Clifford operator and taper qubits.
+
+    The qubit operators for single and double excitations are first generated using the generators of
+    :func:`~.SingleExcitation` and :func:`~.DoubleExcitation` operations. Each of these operators that commutes
+    with all :math:`\mathbb{Z}_2` symmetries of the molecular Hamiltonian are then tranformed using the
+    Clifford operators :math:`U` and then tapered, while rest of the other non-commuting operators are discarded.
+    These new tapered excitation operators can be exponentiated using :func:`~.PauliRot` for building a
+    tapered UCCSD-like circuit ansatze.
```

Comment:
```suggestion
    tapered circuit ansatz.
```
one ansatz, two ansatze